### ISSUE TITLE
CMAC EVP_aes_128_xts() outputs uninitialized memory

### ISSUE THREAD

**Issue by guidovranken** [openssl/openssl#11957]
OSS-Fuzz #22453

```c
#include <openssl/cmac.h>
#define CF_CHECK_EQ(expr, res) if ( (expr) != (res) ) { goto end; }
int main(void)
{
    OPENSSL_init_crypto(OPENSSL_INIT_LOAD_CRYPTO_STRINGS, NULL);
    unsigned char key[32] = {0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20,
        0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0xff};
    unsigned char cleartext[32] = { 0 };
    CMAC_CTX* ctx = CMAC_CTX_new();
    CF_CHECK_EQ(CMAC_Init(ctx, key, sizeof(key), EVP_aes_128_xts(), NULL), 1);
    CF_CHECK_EQ(CMAC_Update(ctx, cleartext, sizeof(cleartext)), 1);
    {
        size_t len = 0;
        uint8_t out[EVP_MAX_MD_SIZE];
        CF_CHECK_EQ(CMAC_Final(ctx, out, &len), 1);
        if ( len ) {
            printf("%02X\n", out[0]);
        }
    }
end:
    return 0;
}
```

```out[0]``` will be uninitialized upon printing it, which you can verify with valgrind.

**Comment by mattcaswell** [comment]
Hmmm. XTS is really not a suitable cipher for use with CMAC. Either we should detect this somehow, and cause CMAC_Init to  fail. Or, possibly, we just document it as something you shouldn't do.

**Comment by guidovranken** [comment]
That's what I suspected. Let me know if you want me to skip CMAC+XTS in Cryptofuzz.

**Comment by mattcaswell** [comment]
Fix in #11972

**Comment by levitte** [comment]
@guidovranken, as you can see from #11972, you should get a proper failure going forward.
